src/mlpack/methods/ann/activation_functions/lazy_logistic_function.hpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file fast_logistic_function.hpp
+ * @author Ngap Wei Tham
+ *
+ * Definition and implementation of the lazy logistic function.This class is same\
+ * as the LogisticFunction, but the function fn do not check upperflow and underflow
+ */
+#ifndef __MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_LAZY_LOGISTIC_FUNCTION_HPP
+#define __MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_LAZY_LOGISTIC_FUNCTION_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The logistic function, defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \frac{1}{1 + e^{-x}} \\
+ * f'(x) &=& f(x) * (1 - f(x)) \\
+ * f^{-1}(y) &=& ln(\frac{y}{1-y})
+ * @f}
+ */
+class LazyLogisticFunction
+{
+  public:
+  /**
+   * Computes the logistic function without checking overflow or underflow
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void fn(const InputVecType& x, OutputVecType& y)
+  {
+    y = (1.0 / (1 + arma::exp(-x)));
```

Comment:
Originally my intention for the overflow checks in the LogisticFunction class was to avoid strange issues during the training process. But I'm intended to remove the checks in the LogisticFunction class, if you agree we can remove the LazyLogisticFunction class and just use the LogisticFunction class.